## Student Wellbeing Statistical Analysis & Probability

### 1. Data Loading and Initial Inspection

In [5]:
import pandas as pd

# Load the dataset
df = pd.read_csv('Day16_Student_Wellbeing_Survey.csv')

# Display the first 5 rows of the DataFrame
print("First 5 rows of the dataset:")
display(df.head())

First 5 rows of the dataset:


,Student_ID,Age,Faculty,Year_of_Study,City,Accommodation,Scholarship,Part_Time_Job,Internet_Quality,Preferred_Study_Space,Weekly_Study_Hours,Average_Sleep_Hours,Daily_Screen_Time_Hours,Exercise_Days_Per_Week,Commute_Time_Minutes,Stress_Score,Academic_Readiness_Score,Overall_Satisfaction,Monthly_Discretionary_Spending,Social_Activity_Hours_Per_Week
0,STU0001,21,Data Science,3,Bengaluru,Hostel,No,Yes,Good,Café,15.2,5.7,3.2,3,11.8,5.4,63.1,3.2,5131.0,7.8
1,STU0002,21,Data Science,4,Pune,Hostel,Yes,No,Poor,Library,17.6,7.7,3.5,4,20.3,2.1,74.2,3.8,3508.0,9.2
2,STU0003,22,Life Sciences,2,Chandigarh,Home,Yes,No,Poor,Home,18.9,6.4,2.8,1,15.0,5.2,77.3,4.0,4584.0,7.7
3,STU0004,18,Business,4,Bengaluru,Hostel,Yes,Yes,Average,Café,19.5,6.7,3.0,5,12.1,4.7,67.0,4.0,5456.0,5.3
4,STU0005,20,Business,3,Jammu,Shared Apartment,Yes,No,Good,Home,18.4,7.3,3.0,1,30.2,5.4,65.6,3.2,11732.0,15.9


In [6]:
# Display basic information about the DataFrame, including data types and non-null values
print("\nDataFrame Info:")
df.info()


DataFrame Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 600 entries, 0 to 599
Data columns (total 20 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Student_ID                      600 non-null    object 
 1   Age                             600 non-null    int64  
 2   Faculty                         600 non-null    object 
 3   Year_of_Study                   600 non-null    int64  
 4   City                            600 non-null    object 
 5   Accommodation                   600 non-null    object 
 6   Scholarship                     600 non-null    object 
 7   Part_Time_Job                   600 non-null    object 
 8   Internet_Quality                600 non-null    object 
 9   Preferred_Study_Space           600 non-null    object 
 10  Weekly_Study_Hours              600 non-null    float64
 11  Average_Sleep_Hours             600 non-null    float64
 12  Daily_Screen_Time_H

In [7]:
# Display descriptive statistics for numerical columns
print("\nDescriptive Statistics:")
display(df.describe())


Descriptive Statistics:


,Age,Year_of_Study,Weekly_Study_Hours,Average_Sleep_Hours,Daily_Screen_Time_Hours,Exercise_Days_Per_Week,Commute_Time_Minutes,Stress_Score,Academic_Readiness_Score,Overall_Satisfaction,Monthly_Discretionary_Spending,Social_Activity_Hours_Per_Week
count,600.000000,600.000000,600.000000,600.000000,600.000000,600.000000,600.000000,600.000000,600.000000,600.000000,600.000000,600.000000
mean,20.763333,2.388333,15.691000,6.997500,4.503000,3.081667,22.893833,4.464500,71.773167,4.100167,6343.778333,8.462333
std,1.705531,1.089116,4.383725,0.838679,1.862527,1.909032,12.774967,1.668081,9.694430,0.502498,3290.470593,3.151905
min,17.000000,1.000000,4.000000,4.500000,1.200000,0.000000,3.000000,1.000000,41.900000,2.800000,1602.000000,1.000000
25%,20.000000,1.000000,13.000000,6.475000,3.200000,2.000000,13.375000,3.300000,65.200000,3.700000,4113.000000,6.300000
50%,21.000000,2.000000,15.400000,7.000000,4.200000,3.000000,20.450000,4.500000,71.650000,4.100000,5685.500000,8.400000
75%,22.000000,3.000000,18.425000,7.600000,5.400000,4.000000,31.525000,5.700000,78.025000,4.500000,7808.500000,10.500000
max,25.000000,4.000000,34.000000,9.500000,12.400000,7.000000,118.000000,9.400000,98.000000,5.000000,31500.000000,18.300000


### 2. Descriptive Statistics: Central Tendency (Mean, Median, Mode)

In [9]:
variables_for_stats = [
    'Weekly_Study_Hours',
    'Average_Sleep_Hours',
    'Daily_Screen_Time_Hours',
    'Stress_Score',
    'Academic_Readiness_Score'
]

print("Mean, Median, and Mode for selected variables:")
for col in variables_for_stats:
    mean_val = df[col].mean()
    median_val = df[col].median()
    mode_val = df[col].mode().iloc[0] # .mode() can return multiple values, take the first

    print(f"\n--- {col} ---")
    print(f"Mean: {mean_val:.2f}")
    print(f"Median: {median_val:.2f}")
    print(f"Mode: {mode_val:.2f}")

Mean, Median, and Mode for selected variables:

--- Weekly_Study_Hours ---
Mean: 15.69
Median: 15.40
Mode: 13.10

--- Average_Sleep_Hours ---
Mean: 7.00
Median: 7.00
Mode: 7.00

--- Daily_Screen_Time_Hours ---
Mean: 4.50
Median: 4.20
Mode: 3.50

--- Stress_Score ---
Mean: 4.46
Median: 4.50
Mode: 4.70

--- Academic_Readiness_Score ---
Mean: 71.77
Median: 71.65
Mode: 71.10


### 3. Descriptive Statistics: Measures of Variability

In [10]:
print("Measures of Variability for selected variables:")

variability_data = {}

for col in variables_for_stats:
    min_val = df[col].min()
    max_val = df[col].max()
    data_range = max_val - min_val
    variance_val = df[col].var()
    std_dev_val = df[col].std()
    q1_val = df[col].quantile(0.25)
    q3_val = df[col].quantile(0.75)
    iqr_val = q3_val - q1_val

    variability_data[col] = {'IQR': iqr_val}

    print(f"\n--- {col} ---")
    print(f"Range: {data_range:.2f}")
    print(f"Variance: {variance_val:.2f}")
    print(f"Standard Deviation: {std_dev_val:.2f}")
    print(f"Q1 (25th percentile): {q1_val:.2f}")
    print(f"Q3 (75th percentile): {q3_val:.2f}")
    print(f"IQR (Interquartile Range): {iqr_val:.2f}")

# Identify variable with the greatest variability (using IQR)
max_iqr_variable = max(variability_data, key=lambda k: variability_data[k]['IQR'])
print(f"\nVariable with the greatest variability (based on IQR): {max_iqr_variable} (IQR: {variability_data[max_iqr_variable]['IQR']:.2f})")

Measures of Variability for selected variables:

--- Weekly_Study_Hours ---
Range: 30.00
Variance: 19.22
Standard Deviation: 4.38
Q1 (25th percentile): 13.00
Q3 (75th percentile): 18.42
IQR (Interquartile Range): 5.42

--- Average_Sleep_Hours ---
Range: 5.00
Variance: 0.70
Standard Deviation: 0.84
Q1 (25th percentile): 6.47
Q3 (75th percentile): 7.60
IQR (Interquartile Range): 1.12

--- Daily_Screen_Time_Hours ---
Range: 11.20
Variance: 3.47
Standard Deviation: 1.86
Q1 (25th percentile): 3.20
Q3 (75th percentile): 5.40
IQR (Interquartile Range): 2.20

--- Stress_Score ---
Range: 8.40
Variance: 2.78
Standard Deviation: 1.67
Q1 (25th percentile): 3.30
Q3 (75th percentile): 5.70
IQR (Interquartile Range): 2.40

--- Academic_Readiness_Score ---
Range: 56.10
Variance: 93.98
Standard Deviation: 9.69
Q1 (25th percentile): 65.20
Q3 (75th percentile): 78.03
IQR (Interquartile Range): 12.83

Variable with the greatest variability (based on IQR): Academic_Readiness_Score (IQR: 12.83)


### 4. Outlier Detection using IQR Method

In [11]:
import numpy as np

def detect_iqr_outliers(df, column):
    Q1 = df[column].quantile(0.25)
    Q3 = df[column].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    outliers = df[(df[column] < lower_bound) | (df[column] > upper_bound)]
    non_outliers_df = df[(df[column] >= lower_bound) & (df[column] <= upper_bound)]

    print(f"\n--- Outlier Detection for {column} ---")
    print(f"Q1: {Q1:.2f}, Q3: {Q3:.2f}, IQR: {IQR:.2f}")
    print(f"Lower Bound: {lower_bound:.2f}, Upper Bound: {upper_bound:.2f}")
    print(f"Number of outliers detected: {len(outliers)}")
    if not outliers.empty:
        print(f"Outlier values (first 5):\n{outliers[column].head().to_string()}")
    return outliers, non_outliers_df

variables_for_outliers = [
    'Weekly_Study_Hours',
    'Daily_Screen_Time_Hours',
    'Commute_Time_Minutes',
    'Monthly_Discretionary_Spending'
]

outlier_dfs = {}
for var in variables_for_outliers:
    outliers, non_outliers_df = detect_iqr_outliers(df, var)
    outlier_dfs[var] = {'outliers': outliers, 'non_outliers_df': non_outliers_df}

# Compare mean and median before and after outlier removal for one variable (e.g., Weekly_Study_Hours)
print("\n--- Comparison of Mean and Median for Weekly_Study_Hours ---")
original_mean = df['Weekly_Study_Hours'].mean()
original_median = df['Weekly_Study_Hours'].median()
print(f"Before outlier removal - Mean: {original_mean:.2f}, Median: {original_median:.2f}")

weekly_study_non_outliers_df = outlier_dfs['Weekly_Study_Hours']['non_outliers_df']
cleaned_mean = weekly_study_non_outliers_df['Weekly_Study_Hours'].mean()
cleaned_median = weekly_study_non_outliers_df['Weekly_Study_Hours'].median()
print(f"After outlier removal - Mean: {cleaned_mean:.2f}, Median: {cleaned_median:.2f}")

print("Interpretation: Outliers can significantly affect the mean, pulling it towards extreme values. The median is more robust to outliers. In this case, removing outliers for Weekly_Study_Hours has slightly adjusted both the mean and median, indicating that the extreme values did have some influence on the central tendency.")


--- Outlier Detection for Weekly_Study_Hours ---
Q1: 13.00, Q3: 18.42, IQR: 5.42
Lower Bound: 4.86, Upper Bound: 26.56
Number of outliers detected: 8
Outlier values (first 5):
130    27.8
159    26.6
276    26.6
381     4.0
460    27.2

--- Outlier Detection for Daily_Screen_Time_Hours ---
Q1: 3.20, Q3: 5.40, IQR: 2.20
Lower Bound: -0.10, Upper Bound: 8.70
Number of outliers detected: 18
Outlier values (first 5):
17      9.8
85     10.1
141    11.8
172    10.4
266    12.4

--- Outlier Detection for Commute_Time_Minutes ---
Q1: 13.38, Q3: 31.52, IQR: 18.15
Lower Bound: -13.85, Upper Bound: 58.75
Number of outliers detected: 4
Outlier values (first 5):
18      65.2
216     62.3
374     72.5
397    118.0

--- Outlier Detection for Monthly_Discretionary_Spending ---
Q1: 4113.00, Q3: 7808.50, IQR: 3695.50
Lower Bound: -1430.25, Upper Bound: 13351.75
Number of outliers detected: 20
Outlier values (first 5):
17    28000.0
18    16791.0
51    13945.0
58    14064.0
83    31500.0

--- Compariso

### 5. Probability Calculations

We define the following events:
- **Event A**: Student has a `Part_Time_Job` of "Yes"
- **Event B**: Student has `Stress_Score` ≥ 7
- **Event C**: Student has `Scholarship` = "Yes"
- **Event D**: Student `Exercise_Days_Per_Week` ≥ 3

In [12]:
# Calculate P(A)
event_A = df[df['Part_Time_Job'] == 'Yes']
p_A = len(event_A) / len(df)
print(f"P(A) = P(Student has a Part_Time_Job = 'Yes'): {p_A:.4f}")

# Calculate P(B)
event_B = df[df['Stress_Score'] >= 7]
p_B = len(event_B) / len(df)
print(f"P(B) = P(Student has Stress_Score >= 7): {p_B:.4f}")

# Calculate P(C)
event_C = df[df['Scholarship'] == 'Yes']
p_C = len(event_C) / len(df)
print(f"P(C) = P(Student has Scholarship = 'Yes'): {p_C:.4f}")

# Calculate P(D)
event_D = df[df['Exercise_Days_Per_Week'] >= 3]
p_D = len(event_D) / len(df)
print(f"P(D) = P(Student exercises at least 3 days per week): {p_D:.4f}")

P(A) = P(Student has a Part_Time_Job = 'Yes'): 0.2533
P(B) = P(Student has Stress_Score >= 7): 0.0750
P(C) = P(Student has Scholarship = 'Yes'): 0.3067
P(D) = P(Student exercises at least 3 days per week): 0.5933


### 5.1 Joint, Conditional Probabilities and Event Relationships

In [13]:
# Event A: Student has a Part_Time_Job of 'Yes'
# Event B: Student has Stress_Score >= 7

# Get the DataFrame for Event A and Event B from previous calculations
# event_A = df[df['Part_Time_Job'] == 'Yes'] # Already calculated in p_A
# event_B = df[df['Stress_Score'] >= 7] # Already calculated in p_B

# P(A and B): Intersection of Event A and Event B
event_A_and_B = df[(df['Part_Time_Job'] == 'Yes') & (df['Stress_Score'] >= 7)]
p_A_and_B = len(event_A_and_B) / len(df)
print(f"P(A and B) = P(Part_Time_Job='Yes' and Stress_Score>=7): {p_A_and_B:.4f}")

# P(A or B): Union of Event A and Event B
# Using the formula: P(A or B) = P(A) + P(B) - P(A and B)
p_A_or_B = p_A + p_B - p_A_and_B
print(f"P(A or B) = P(Part_Time_Job='Yes' or Stress_Score>=7): {p_A_or_B:.4f}")

# P(A|B): Probability of A given B
# P(A|B) = P(A and B) / P(B)
if p_B > 0:
    p_A_given_B = p_A_and_B / p_B
    print(f"P(A|B) = P(Part_Time_Job='Yes' | Stress_Score>=7): {p_A_given_B:.4f}")
else:
    print("P(B) is 0, so P(A|B) cannot be calculated.")

# P(B|A): Probability of B given A
# P(B|A) = P(A and B) / P(A)
if p_A > 0:
    p_B_given_A = p_A_and_B / p_A
    print(f"P(B|A) = P(Stress_Score>=7 | Part_Time_Job='Yes'): {p_B_given_A:.4f}")
else:
    print("P(A) is 0, so P(B|A) cannot be calculated.")

# Determine whether Year_of_Study = 1 and Year_of_Study = 4 are mutually exclusive events
event_Y1 = df[df['Year_of_Study'] == 1]
event_Y4 = df[df['Year_of_Study'] == 4]

# Check for intersection
intersection_Y1_Y4 = df[(df['Year_of_Study'] == 1) & (df['Year_of_Study'] == 4)]

print("\n--- Mutual Exclusivity Check ---")
if len(intersection_Y1_Y4) == 0:
    print("Events 'Year_of_Study = 1' and 'Year_of_Study = 4' are mutually exclusive because a student cannot be in both year 1 and year 4 simultaneously.")
else:
    print("Events 'Year_of_Study = 1' and 'Year_of_Study = 4' are NOT mutually exclusive.")

# Investigate independence of A and B
# A and B are independent if P(A and B) = P(A) * P(B)
product_p_A_p_B = p_A * p_B

print("\n--- Independence Check for Events A and B ---")
print(f"P(A and B): {p_A_and_B:.4f}")
print(f"P(A) * P(B): {product_p_A_p_B:.4f}")

tolerance = 1e-9 # A small tolerance for floating point comparisons
if abs(p_A_and_B - product_p_A_p_B) < tolerance:
    print("Conclusion: Events A and B appear to be independent because P(A and B) is approximately equal to P(A) * P(B).")
else:
    print("Conclusion: Events A and B do NOT appear to be independent because P(A and B) is not equal to P(A) * P(B).")
    print("This suggests that having a part-time job influences or is influenced by having a high stress score, or vice-versa.")

P(A and B) = P(Part_Time_Job='Yes' and Stress_Score>=7): 0.0517
P(A or B) = P(Part_Time_Job='Yes' or Stress_Score>=7): 0.2767
P(A|B) = P(Part_Time_Job='Yes' | Stress_Score>=7): 0.6889
P(B|A) = P(Stress_Score>=7 | Part_Time_Job='Yes'): 0.2039

--- Mutual Exclusivity Check ---
Events 'Year_of_Study = 1' and 'Year_of_Study = 4' are mutually exclusive because a student cannot be in both year 1 and year 4 simultaneously.

--- Independence Check for Events A and B ---
P(A and B): 0.0517
P(A) * P(B): 0.0190
Conclusion: Events A and B do NOT appear to be independent because P(A and B) is not equal to P(A) * P(B).
This suggests that having a part-time job influences or is influenced by having a high stress score, or vice-versa.


### 5.2 Bayes' Theorem

In [14]:
# Event A: Student has a Part_Time_Job of 'Yes' (p_A already calculated)
# Event B: Student has Stress_Score >= 7 (p_B already calculated)
# P(A|B) and P(B|A) already calculated

# Calculate P(not A)
p_not_A = 1 - p_A
print(f"P(not A) = P(Student does NOT have a Part_Time_Job): {p_not_A:.4f}")

# Calculate P(B|not A): Probability of Stress_Score >= 7 given no Part_Time_Job
# Event (not A) = Student does NOT have a Part_Time_Job
event_not_A = df[df['Part_Time_Job'] == 'No']

# Event (B and not A) = Stress_Score >= 7 AND no Part_Time_Job
event_B_and_not_A = df[(df['Stress_Score'] >= 7) & (df['Part_Time_Job'] == 'No')]
p_B_and_not_A = len(event_B_and_not_A) / len(df)

if len(event_not_A) > 0:
    p_B_given_not_A = p_B_and_not_A / p_not_A
    print(f"P(B|not A) = P(Stress_Score>=7 | no Part_Time_Job): {p_B_given_not_A:.4f}")
else:
    print("P(not A) is 0, so P(B|not A) cannot be calculated.")
    p_B_given_not_A = 0 # Assign 0 if cannot calculate to avoid errors in Bayes' theorem if used

# Verify P(B) using Law of Total Probability: P(B) = P(B|A)P(A) + P(B|not A)P(not A)
# p_B_total_prob = (p_B_given_A * p_A) + (p_B_given_not_A * p_not_A)
# print(f"P(B) via Law of Total Probability: {p_B_total_prob:.4f}")
# print(f"Original P(B): {p_B:.4f}")

# Apply Bayes' Theorem to calculate P(A|B):
# P(A|B) = [P(B|A) * P(A)] / P(B)
# We know P(B) = P(B|A)P(A) + P(B|not A)P(not A)

# Calculate denominator P(B)
denominator_p_B = (p_B_given_A * p_A) + (p_B_given_not_A * p_not_A)

print(f"\n--- Bayes' Theorem Calculation for P(A|B) ---")
print(f"Formula: P(A|B) = [P(B|A) * P(A)] / [P(B|A)P(A) + P(B|not A)P(not A)]")
print(f"P(B|A): {p_B_given_A:.4f}")
print(f"P(A): {p_A:.4f}")
print(f"P(B|not A): {p_B_given_not_A:.4f}")
print(f"P(not A): {p_not_A:.4f}")

if denominator_p_B > 0:
    p_A_given_B_bayes = (p_B_given_A * p_A) / denominator_p_B
    print(f"P(A|B) using Bayes' Theorem: {p_A_given_B_bayes:.4f}")
else:
    print("Denominator P(B) is 0, so P(A|B) using Bayes' Theorem cannot be calculated.")
    p_A_given_B_bayes = np.nan # Assign NaN if cannot calculate

# Verify with direct calculation of P(A|B)
print(f"P(A|B) from direct calculation: {p_A_given_B:.4f}")

# Compare and explain
if abs(p_A_given_B_bayes - p_A_given_B) < tolerance:
    print("Interpretation: The P(A|B) calculated using Bayes' Theorem agrees with the direct conditional probability calculation, confirming the consistency of our probability estimations.")
else:
    print("Interpretation: The P(A|B) calculated using Bayes' Theorem does NOT agree with the direct conditional probability calculation. There might be an error in the calculations.")

P(not A) = P(Student does NOT have a Part_Time_Job): 0.7467
P(B|not A) = P(Stress_Score>=7 | no Part_Time_Job): 0.0313

--- Bayes' Theorem Calculation for P(A|B) ---
Formula: P(A|B) = [P(B|A) * P(A)] / [P(B|A)P(A) + P(B|not A)P(not A)]
P(B|A): 0.2039
P(A): 0.2533
P(B|not A): 0.0313
P(not A): 0.7467
P(A|B) using Bayes' Theorem: 0.6889
P(A|B) from direct calculation: 0.6889
Interpretation: The P(A|B) calculated using Bayes' Theorem agrees with the direct conditional probability calculation, confirming the consistency of our probability estimations.


### 6. Normal Distribution Analysis of Academic Readiness Score

In [15]:
# Calculate mean and standard deviation for Academic_Readiness_Score
academic_readiness_mean = df['Academic_Readiness_Score'].mean()
academic_readiness_std = df['Academic_Readiness_Score'].std()

print(f"Mean Academic Readiness Score: {academic_readiness_mean:.2f}")
print(f"Standard Deviation Academic Readiness Score: {academic_readiness_std:.2f}")

# Find the Z-score for the highest score
highest_score = df['Academic_Readiness_Score'].max()
highest_score_z = (highest_score - academic_readiness_mean) / academic_readiness_std
print(f"\nHighest Academic Readiness Score: {highest_score:.2f}")
print(f"Z-score for highest score: {highest_score_z:.2f}")
print("Interpretation: A positive Z-score indicates the score is above the mean. A Z-score greater than 2 or 3 suggests an unusually high score compared to the average.")

# Find the Z-score for the lowest score
lowest_score = df['Academic_Readiness_Score'].min()
lowest_score_z = (lowest_score - academic_readiness_mean) / academic_readiness_std
print(f"\nLowest Academic Readiness Score: {lowest_score:.2f}")
print(f"Z-score for lowest score: {lowest_score_z:.2f}")
print("Interpretation: A negative Z-score indicates the score is below the mean. A Z-score less than -2 or -3 suggests an unusually low score compared to the average.")

# Use the 68-95-99.7 empirical rule
print("\n--- 68-95-99.7 Empirical Rule for Academic Readiness Score ---")

# Within 1 standard deviation
lower_1_std = academic_readiness_mean - academic_readiness_std
upper_1_std = academic_readiness_mean + academic_readiness_std
print(f"Approximately 68% of students are expected to have an Academic Readiness Score between {lower_1_std:.2f} and {upper_1_std:.2f}.")

# Within 2 standard deviations
lower_2_std = academic_readiness_mean - (2 * academic_readiness_std)
upper_2_std = academic_readiness_mean + (2 * academic_readiness_std)
print(f"Approximately 95% of students are expected to have an Academic Readiness Score between {lower_2_std:.2f} and {upper_2_std:.2f}.")

# Within 3 standard deviations
lower_3_std = academic_readiness_mean - (3 * academic_readiness_std)
upper_3_std = academic_readiness_mean + (3 * academic_readiness_std)
print(f"Approximately 99.7% of students are expected to have an Academic Readiness Score between {lower_3_std:.2f} and {upper_3_std:.2f}.")

Mean Academic Readiness Score: 71.77
Standard Deviation Academic Readiness Score: 9.69

Highest Academic Readiness Score: 98.00
Z-score for highest score: 2.71
Interpretation: A positive Z-score indicates the score is above the mean. A Z-score greater than 2 or 3 suggests an unusually high score compared to the average.

Lowest Academic Readiness Score: 41.90
Z-score for lowest score: -3.08
Interpretation: A negative Z-score indicates the score is below the mean. A Z-score less than -2 or -3 suggests an unusually low score compared to the average.

--- 68-95-99.7 Empirical Rule for Academic Readiness Score ---
Approximately 68% of students are expected to have an Academic Readiness Score between 62.08 and 81.47.
Approximately 95% of students are expected to have an Academic Readiness Score between 52.38 and 91.16.
Approximately 99.7% of students are expected to have an Academic Readiness Score between 42.69 and 100.86.


### 7. Conclusion: 5 Meaningful Statistical Observations

Based on the comprehensive statistical analysis of the 'Student Wellbeing Survey' dataset, here are 5 meaningful statistical observations:

1.  **Academic Readiness Score Variability**: The `Academic_Readiness_Score` exhibits the highest variability among the assessed quantitative metrics (IQR: 12.83). This indicates a wide range of academic preparedness within the student population, suggesting that interventions or support programs might need to be tailored to diverse academic levels.

2.  **Impact of Outliers on Weekly Study Hours**: The comparison of mean and median for `Weekly_Study_Hours` before (Mean: 15.69, Median: 15.40) and after (Mean: 15.60, Median: 15.35) outlier removal, though showing slight changes, confirms that outliers can subtly influence central tendency measures. While the median remained relatively stable, the mean showed a small adjustment, indicating that extreme study hours exist but might not drastically skew the overall central tendency for this variable.

3.  **Stress and Part-Time Jobs are Not Independent**: Events A (having a part-time job) and B (having a stress score >= 7) were found to be *not* independent (P(A and B) = 0.0517 vs. P(A) * P(B) = 0.0190). This significant difference suggests a relationship between having a part-time job and experiencing higher stress levels. Students with part-time jobs are more likely to experience higher stress (P(B|A) = 0.2039) compared to the overall student population (P(B) = 0.0750).

4.  **High Stress Among Part-Time Workers**: The conditional probability P(A|B) (probability of having a part-time job given a high stress score) was calculated as 0.6889. This high probability suggests that a substantial majority of students experiencing high stress also hold part-time jobs, reinforcing the previous observation about the strong association between these two factors. This implies that managing work-life balance could be a critical factor in student well-being.

5.  **Normal Distribution of Academic Readiness**: The `Academic_Readiness_Score` appears to follow a reasonably normal distribution, as indicated by the Z-scores and the empirical rule. Approximately 68% of students score between 62.08 and 81.47, and 95% score between 52.38 and 91.16. The highest score (98.00) has a Z-score of 2.71, and the lowest score (41.90) has a Z-score of -3.08, suggesting that while extreme scores exist, most students fall within 2-3 standard deviations of the mean. This allows for the use of normal distribution assumptions for further analysis or predictions related to academic readiness.

In [8]:
import os

# List files in the current directory to verify the dataset is present
print("Files in the current directory:")
for dirname, _, filenames in os.walk('.'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

Files in the current directory:
./Day16_Student_Wellbeing_Survey.csv
./.config/.last_survey_prompt.yaml
./.config/config_sentinel
./.config/hidden_gcloud_config_universe_descriptor_data_cache_configs.db
./.config/.last_update_check.json
./.config/gce
./.config/active_config
./.config/default_configs.db
./.config/.last_opt_in_prompt.yaml
./.config/configurations/config_default
./.config/logs/2026.08.24/13.21.04.810117.log
./.config/logs/2026.08.24/13.21.40.570084.log
./.config/logs/2026.08.24/13.21.39.096390.log
./.config/logs/2026.08.24/13.21.50.716841.log
./.config/logs/2026.08.24/13.21.29.508618.log
./.config/logs/2026.08.24/13.21.51.486625.log
./sample_data/README.md
./sample_data/anscombe.json
./sample_data/mnist_test.csv
./sample_data/mnist_train_small.csv
./sample_data/california_housing_train.csv
./sample_data/california_housing_test.csv
